site/sfguides/src/ingest_data_from_pubsub_to_snowflake_with_apache_beam/beam.md

```diff
@@ -0,0 +1,541 @@
+<!-- INSERT path:./setup.sh text_id:create_user lang:bash-->
+```bash
+snowsql -c "${SNOWSQL_CONN}" -q "
+  CREATE OR REPLACE ROLE ${SNOWFLAKE_ROLE};
+  CREATE OR REPLACE USER ${SNOWFLAKE_USERNAME} DEFAULT_ROLE=${SNOWFLAKE_ROLE}, DEFAULT_WAREHOUSE=${SNOWFLAKE_WAREHOUSE} RSA_PUBLIC_KEY='${SNOWFLAKE_PUB_KEY}';
+
+  GRANT ROLE ${SNOWFLAKE_ROLE} TO USER ${SNOWFLAKE_USERNAME}
+"
+```
+<!-- END -->
+
+You can use SnowSQL to validate the service user's configuration. To do it, run:
+<!-- INSERT path:./setup.sh text_id:verify_user lang:bash-->
+```bash
+SNOWSQL_PRIVATE_KEY_PASSPHRASE="${SNOWFLAKE_PRIVATE_KEY_PASSPHASE}" \
+   snowsql \
+   --accountname "$(echo "${SNOWFLAKE_SERVER_NAME}" | cut -d "." -f 1-2)" \
+   --username "${SNOWFLAKE_USERNAME}" \
+   --dbname "${SNOWFLAKE_DATABASE}" \
+   --schemaname "${SNOWFLAKE_SCHEMA}" \
+   --warehouse "${SNOWFLAKE_WAREHOUSE}" \
+   --rolename "${SNOWFLAKE_ROLE}" \
+   --private-key-path "rsa_key.p8" \
+   --query 'SELECT CURRENT_ROLE(), CURRENT_USER()';
+```
+<!-- END -->
+
+If you run into difficulties, check out the article [Key Pair Authentication & Key Pair Rotation
+](https://docs.snowflake.com/en/user-guide/key-pair-auth.html) in the Snowflake documentation.
+
+<!-- ------------------------ -->
+## Setting up database, schema in Snowflake
+
+Set a variables that describe the Snowflake account and tables as in the example below:
+<!-- INSERT path:./.env.example text_id:connection lang:bash-->
+```bash
+SNOWFLAKE_SERVER_NAME="XXX.snowflakecomputing.com"
+SNOWFLAKE_DATABASE="DEV_XXX_BEAM"
+SNOWFLAKE_SCHEMA="DEV_XXX"
+```
+<!-- END -->
+where:
+- `SNOWFLAKE_SERVER_NAME` should specify the name of the server you will connect to. It must to end with `.snowflakecomputing.com`.
+- `SNOWFLAKE_DATABASE` - the name of the new database to be created
+- `SNOWFLAKE_SCHEMA` - the name of the new schema to be created
+
+To create a new database and schema and grant privilege, run:
+<!-- INSERT path:./setup.sh text_id:create_database lang:bash-->
+```bash
+snowsql -c "${SNOWSQL_CONN}" -q "
+  CREATE OR REPLACE DATABASE ${SNOWFLAKE_DATABASE};
+  CREATE OR REPLACE SCHEMA ${SNOWFLAKE_DATABASE}.${SNOWFLAKE_SCHEMA};
+
+  GRANT USAGE ON DATABASE ${SNOWFLAKE_DATABASE} TO ROLE ${SNOWFLAKE_ROLE};
+  GRANT USAGE ON SCHEMA ${SNOWFLAKE_DATABASE}.${SNOWFLAKE_SCHEMA} TO ROLE ${SNOWFLAKE_ROLE};
+"
+```
+<!-- END -->
+
+<!-- ------------------------ -->
+## Setting up a bucket in GCP, stage and tables in Snowflake
+
+Let's start with setting up a few variables.
+<!-- INSERT path:./.env.example text_id:table lang:bash-->
+```bash
+DATAFLOW_BUCKET="sfc-pubsub-to-snowflake-dataflow"
+SNOWFLAKE_STORAGE_INTEGRATION="DEV_XXX_BEAM_STORAGE_INTEGRATION"
+SNOWFLAKE_STAGE="DEV_XXX_BEAM_STAGE"
+PIPELINE_SNOWFLAKE_OUTPUT_TABLE="PUBSUB_MESSAGES"
+```
+<!-- END -->
+
+where:
+- `DATAFLOW_BUCKET` - the name of the new bucket to be created. It will be used as the staging area. Every bucket name are globally unique, so you will have to update value.
```

Comment:
`so you will have to update value.` I am not sure what do you mean - why and where it will be necessary to update the value?